test/test_CommandAndStateMessageParser.cpp

```diff
@@ -376,6 +375,38 @@ TEST_F(MessageParserTest, it_returns_the_camera_head_state)
     ASSERT_EQ(expected.leak, actual.leak);
 }
 
+TEST_F(MessageParserTest, it_returns_the_camera_head_tilt_state_in_rbs_form)
+{
+    auto parser = getMessageParser();
+    Json::Value camera_head;
+    camera_head["payload"]["devices"]["revolution_id123"]["model"] = 13;
+    camera_head["payload"]["devices"]["revolution_id123"]["cameraHead"]["light"]
+               ["intensity"] = 40;
+    camera_head["payload"]["devices"]["revolution_id123"]["cameraHead"]["lasers"]
+               ["enabled"] = true;
+    camera_head["payload"]["devices"]["revolution_id123"]["cameraHead"]["leak"] = false;
+    camera_head["payload"]["devices"]["revolution_id123"]["cameraHead"]["tilt"]
+               ["position"] = 90;
+    camera_head["payload"]["devices"]["revolution_id123"]["cameraHead"]
+               ["tiltMotorDiagnostics"]["overcurrent"] = true;
+    camera_head["payload"]["devices"]["revolution_id123"]["cameraHead"]
+               ["tiltMotorDiagnostics"]["rpm"] = 20;
+    camera_head["payload"]["devices"]["revolution_id123"]["cameraHead"]
+               ["tiltMotorDiagnostics"]["pwm"] = 80;
+    camera_head["payload"]["devices"]["revolution_id123"]["cameraHead"]
+               ["tiltMotorDiagnostics"]["current"] = 60;
+
+    Json::FastWriter writer;
+    string errors;
+    parser.parseJSONMessage(writer.write(camera_head).c_str(), errors);
+    auto actual = parser.getCameraHeadTiltMotorStateRBS("revolution_id123");
+    base::samples::RigidBodyState expected;
+    expected.orientation = base::AngleAxisd(2.826, base::Vector3d::UnitZ());
+    ASSERT_EQ(getRoll(expected.orientation), 0.0);
+    ASSERT_EQ(getPitch(expected.orientation), 0.0);
+    ASSERT_EQ(getYaw(expected.orientation), 2.826);
```

Comment:
I didn't get it. shouldn't you be testing the camera_head["payload"]["devices"]["revolution_id123"]["cameraHead"]["tilt"]["position"]?

Reply:
Yep. I should